File: frontend/src/api/notifications.ts
import { apiClient } from "./client";

export interface Notification {
  id: number;
  signal_type: string;
  entity_id: number;
  title: string;
  detail: string | null;
  first_seen_at: string;
  read_at: string | null;
  is_new: boolean;
}

export const listNotifications = async (): Promise<Notification[]> =>
  (await apiClient.get<Notification[]>("/notifications")).data;

export const markNotificationRead = async (id: number): Promise<Notification> =>
  (await apiClient.post<Notification>(`/notifications/${id}/read`)).data;

File: backend/app/schemas/notifications.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    signal_type: str
    entity_id: int
    title: str
    detail: str | None
    first_seen_at: datetime
    read_at: datetime | None
    # Не видел ни разу с момента появления — для подсветки в списке
    # отдельно от общего бейджа "непрочитанных" на самом колокольчике.
    is_new: bool
